exercises/bracket-push/example.rs

```diff
@@ -53,23 +68,18 @@ impl Brackets {
     }
 
     pub fn are_balanced(&self) -> bool {
-        let mut unmatched: Vec<char> = Vec::new();
+        let mut unclosed: Vec<char> = Vec::new();
 
         for &bracket in self.raw_brackets.iter() {
-            if let Some(last_unmatched) = unmatched.pop() {
-                if let Some(closing_bracket) = self.pairs.closer_for(&last_unmatched) {
-                    if *closing_bracket != bracket {
-                        unmatched.push(last_unmatched);
-                        unmatched.push(bracket);
-                    }
-                } else {
-                    unmatched.push(bracket);
+            if let Some(last_unclosed) = unclosed.pop() {
+                for c in self.pairs.unmatched(last_unclosed, bracket) {
```

Comment:
take a look at whether this can be expressed as an [`extend`](https://doc.rust-lang.org/std/iter/trait.Extend.html) which I think cn work because the docs say `impl<T> Extend<T> for Vec<T>`.

the idea of letting the `pairs`  determine what's unmatched seems OK.